tests/test_scalar_optimizers.py

```diff
@@ -23,22 +24,54 @@
 )
 
 
+# Define command line flags
+flags.DEFINE_string("device", "cpu", "Device to run tests on: 'cpu', 'cuda', or 'auto'")
+flags.DEFINE_integer("seed", 42, "Random seed for reproducible tests")
+flags.DEFINE_boolean("skip_gpu_tests", False, "Skip GPU tests even if CUDA is available")
+
+FLAGS = flags.FLAGS
+
+
 # Base class for tests requiring seeding for determinism
 class BaseTestCase(parameterized.TestCase):
     def setUp(self):
-        """Set random seed before each test."""
-        # Set seed for PyTorch
-        torch.manual_seed(42)
+        """Set random seed and device before each test."""
+        # Set seed for PyTorch (using seed from flags)
+        torch.manual_seed(FLAGS.seed)
         # Set seed for CUDA if available
         if torch.cuda.is_available():
-            torch.cuda.manual_seed_all(42)
+            torch.cuda.manual_seed_all(FLAGS.seed)
+
+        # Set up device based on flags
+        self.device = self._get_test_device()
+
+    def _get_test_device(self):
+        """Get the device to use for testing based on flags."""
+        if FLAGS.device == "auto":
+            return "cuda" if torch.cuda.is_available() and not FLAGS.skip_gpu_tests else "cpu"
+        elif FLAGS.device == "cuda":
+            if not torch.cuda.is_available():
+                self.skipTest("CUDA not available")
+            if FLAGS.skip_gpu_tests:
```

Comment:
If all tests in a test case is suppose to run with CUDA, don't skip if no GPU is found, let it fail. Because it means it launched to the wrong machine or machine has issue.
Silent skipping and passing test is very bad behavior.